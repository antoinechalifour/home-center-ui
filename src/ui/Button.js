import styled from 'styled-components'

const Button = styled.button`
  font-size: inherit;
  font-family: inherit;
  background: none;
  border: none;
  cursor: pointer;
  text-transform: uppercase;
  padding: 4px 8px;
  border-radius: 4px;
  transition: letter-spacing .25s ease-in;

  :hover {
    letter-spacing: 1px;
  }

  ${({ theme, primary }) => primary && `
    background: ${theme.colors.primary};
    color: #fff;
  `}
`

export default Button
