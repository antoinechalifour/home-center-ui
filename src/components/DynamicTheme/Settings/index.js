import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import Card, * as card from 'ui/Card'
import Button from 'ui/Button'
import ColorPicker from 'ui/ColorPicker'

export default function Settings ({ theme, onColorChange, onCommit, onReset }) {
  const colorsSettings = theme.colors

  return (
    <Wrapper>
      <Content>
        {Object.keys(colorsSettings).map(setting => {
          const color = colorsSettings[setting]

          return (
            <Color key={setting}>
              <ColorPicker
                value={color}
                onChange={value => onColorChange(setting, value)}
              />
              <span>{setting}</span>
            </Color>
          )
        })}

        <Actions>
          <Button onClick={onReset}>Reset theme</Button>
          <Button primary onClick={onCommit}>Validate</Button>
        </Actions>
      </Content>
    </Wrapper>
  )
}

Settings.propTypes = {
  theme: PropTypes.shape({
    colors: PropTypes.object.isRequired
  }).isRequired,
  onColorChange: PropTypes.func.isRequired,
  onCommit: PropTypes.func.isRequired,
  onReset: PropTypes.func.isRequired
}

const Wrapper = styled(Card)`
  position: relative;
  width: 100%;
  font-size: 75%;
`

const Content = styled(card.Content)`
  display: flex;
  flex-direction: column;

  > div {
    padding: 12px;
  }

  @media (min-width: 860px) {
    flex-direction: row;
    align-items: center;
    justify-content: center;
  }
`

const Color = styled.div`
  display: flex;
  align-items: center;

  > :first-child {
    display: inline-flex;
    margin-right: 12px;
  }
`

const Actions = styled.div`
  display: flex;
  flex-direction: column;

  button {
    flex: 1;
  }

  button + button {
    margin-top: 4px;
  }
`
