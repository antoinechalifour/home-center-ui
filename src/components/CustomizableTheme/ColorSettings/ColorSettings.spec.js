import React from 'react'
import { shallow } from 'enzyme'
import Button from 'ui/Button'
import Settings from '.'

const theme = {
  colors: {
    background: '#000',
    text: '#fff'
  }
}

describe('components/CustomizableTheme/Settings', () => {
  it('Should render correctly', () => {
    const props = {
      theme,
      changeColor: jest.fn(),
      commit: jest.fn(),
      reset: jest.fn()
    }
    const wrapper = shallow(<Settings {...props} />)

    expect(wrapper).toMatchSnapshot()
  })

  it('Should call changeColor when the settings change', () => {
    const props = {
      theme,
      changeColor: jest.fn(),
      commit: jest.fn(),
      reset: jest.fn()
    }
    const wrapper = shallow(<Settings {...props} />)

    wrapper.find('ColorPicker').at(1).simulate('change', 'blue')

    expect(props.changeColor.mock.calls.length).toBe(1)
    expect(props.changeColor.mock.calls[0][0]).toBe('text')
    expect(props.changeColor.mock.calls[0][1]).toBe('blue')
  })

  it('Should call reset when the reset button is clicked', () => {
    const props = {
      theme,
      changeColor: jest.fn(),
      commit: jest.fn(),
      reset: jest.fn()
    }
    const wrapper = shallow(<Settings {...props} />)

    wrapper.find(Button).at(0).simulate('click')

    expect(props.reset.mock.calls.length).toBe(1)
  })

  it('Should call reset when the reset button is clicked', () => {
    const props = {
      theme,
      changeColor: jest.fn(),
      commit: jest.fn(),
      reset: jest.fn()
    }
    const wrapper = shallow(<Settings {...props} />)

    wrapper.find(Button).at(1).simulate('click')

    expect(props.commit.mock.calls.length).toBe(1)
  })
})
