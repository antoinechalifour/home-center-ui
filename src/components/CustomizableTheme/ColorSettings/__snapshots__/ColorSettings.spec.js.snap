// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/CustomizableTheme/Settings Should render correctly 1`] = `
<styled.div>
  <styled.div
    key="background"
  >
    <ColorPicker
      onChange={[Function]}
      value="#000"
    />
    <span>
      background
    </span>
  </styled.div>
  <styled.div
    key="text"
  >
    <ColorPicker
      onChange={[Function]}
      value="#fff"
    />
    <span>
      text
    </span>
  </styled.div>
  <styled.div>
    <styled.button
      onClick={[Function]}
    >
      Reset theme
    </styled.button>
    <styled.button
      onClick={[Function]}
      primary={true}
    >
      Validate
    </styled.button>
  </styled.div>
</styled.div>
`;
